tiledb/sm/query/ast/query_ast.h

```diff
@@ -355,13 +398,41 @@ class ASTNodeExpr : public ASTNode {
       , combination_op_(c_op) {
   }
 
+  /**
+   * @brief Copy constructor.
+   */
+  ASTNodeExpr(const ASTNodeExpr& rhs)
+      : combination_op_(rhs.combination_op_) {
+    for (auto& node : rhs.nodes_) {
+      if (node->is_expr()) {
+        nodes_.emplace_back(
+            tdb_new(ASTNodeExpr, *dynamic_cast<ASTNodeExpr*>(node.get())));
```

Comment:
`dynamic_cast` is slow and can be a hit on performance , let's find another way to do this. 
Let me think of an alternative design.